### PR TITLE
README: Add plugin benchmark results

### PR DESCRIPTION
**Pull request by ashutosh-narkar**
This commit adds the benchmarking results obtained
by load testing the plugin under different conditions
and capturing the latency by percentile distribution.

It also provides hints to debug performance issues.

Fixes open-policy-agent/opa#2806

Signed-off-by: Ashutosh Narkar <anarkar4387@gmail.com>

### CODE REVIEW

**Review by srenatus — APPROVED** [review]
👏 Nicely done! LGTM.

The only nitpick I had while reading was that the y axes are different for the different plots, making it difficult to compare envoy_grpc vs google_grpc -- but then you do compare them in the same plot! So, that nitpick doesn't apply anymore.

File: README.md [hunk 1]
```diff
@@ -745,6 +745,188 @@ This section provides examples of interacting with the Envoy External Authorizat
...
+![hist_100](./docs/hist_google_grpc_100.png)
+
+The above four scenarios are replicated to measure the latency distribution now by sending *1000 requests/sec*
+to the sample application. The following graph captures this result.
+
+![hist_1000](./docs/hist_google_grpc_1000.png)
+
+#### OPA Benchmarks
+
+The table below captures the `gRPC Server Handler` and `OPA Evaluation` time with [Envoy External
```
Comment (tsandall) [inline README.md:854]: Do we explain what these two components mean anywhere? It's probably worth mentioning that the `gRPC Server Handler` includes `OPA Evaluation`. In other words, `OPA Evaluation` is the time taken to evaluate the policy. `gRPC Server Handler` is the total time taken to prepare the input for the policy, evaluate the policy (`OPA Evaluation`), and prepare the result.

**Review by tsandall — COMMENTED** [review]
This looks great. A couple minor comments.

File: README.md [hunk 2]
```diff
@@ -745,6 +745,188 @@ This section provides examples of interacting with the Envoy External Authorizat
...
+
+| Number of Requests per sec | 75% | 90% | 95% | 99% | 99.9% | 99.99% | Mean | Median |
+|:--------------------------:|:---:|:---:|:---:|:---:|:-----:|:------:|:----:|:------:|
+| 10 | 467122.5 | 571028.1000000001 | 579272 | 579272 | 579272 | 579272 | 437282.2 | 437367 |
+| 100 | 374714 | 439620.5 | 502146.69999999966 | 736478.7799999997 | 737025 | 737025 | 326840.78 | 307902 |
+| 1000 | 251582.25| 386282.89999999997 | 526037.849999999 | 1.5134571300000004e+06 | 5.627604689000038e+06 | 5.629248e+06 | 269490.797 | 202471.5 |
+| 2000 | 223174.5 | 463464.60000000003 | 1.4615589999999993e+06 | 3.4564649000000022e+06 | 8.251585116000005e+06 | 8.289024e+06 | 345520.33171206224 | 168145 |
+| 3000 | 253609 | 660672.8 | 2.1717189499999997e+06 | 6.878808970000023e+06 | 1.4992017226400027e+08 | 1.52030931e+08 | 800997.785992218 | 163617.5 |
+| 4000 | 266848 | 625534.5000000005 | 1.9575684499999855e+06 | 1.897046686000001e+07 | 1.5220940807700014e+08 | 1.53406983e+08 | 1.0105490768482491e+06 | 162288.5 |
+| 5000 | 302344 | 1.1532631000000008e+06 | 2.2830179499999997e+06 | 6.149344860000005e+06 | 1.3199011520000002e+08 | 1.32097885e+08 | 984640.5642023346 | 170703.5 |
```
[the remark this answers is not in the document]
Comment (ashutosh-narkar) [reply README.md]: Done

File: README.md (hunk 1 above)
Comment (ashutosh-narkar) [reply README.md:854]: Done

**Pull request merged**